File: src/main/java/com/bibireden/data_attributes/mutable/MutableSimpleRegistry.java
package com.bibireden.data_attributes.mutable;

import net.minecraft.util.Identifier;
import net.minecraft.registry.Registry;

public interface MutableSimpleRegistry<T> {
    // Default method to remove cached identifiers from a registry
    default void data_attributes$removeCachedIds(Registry<T> registry) {}

    // Default method to cache an identifier
    default void data_attributes$cacheID(Identifier id) {}
}

File: src/main/kotlin/com/bibireden/data_attributes/mutable/MutableSimpleRegistry.kt
package com.bibireden.data_attributes.mutable

import net.minecraft.registry.Registry
import net.minecraft.util.Identifier

interface MutableSimpleRegistry<T : Any> {
    /** Remove cached identifiers from a registry. */
    fun removeCachedIds(registry: Registry<T>)
    /** Caches an identifier to the registry. */
    fun cacheId(id: Identifier)
}